code/src/App.js

```diff
@@ -1,9 +1,75 @@
-import React from 'react'
+import React, { useEffect, useState } from "react";
+
+import { NewPost } from "./components/NewPost";
+import Posts from "./components/Posts";
+import LoadingScreen from "./components/LoadingScreen";
+
+import { API_URL, LIKES_URL } from "./utils/urls";
 
 export const App = () => {
+  const [thoughts, setThoughts] = useState([]);
+  const [newThought, setNewThought] = useState("");
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    fetchThoughts();
+  }, []);
+
+  const fetchThoughts = () => {
+    setLoading(true);
+    fetch(API_URL)
+      .then((res) => res.json())
+      .then((data) => setThoughts(data))
+      .finally(() => setLoading(false));
+  };
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ message: newThought }),
+    };
+
+    fetch(API_URL, options)
+      .then((res) => res.json())
+      .then((data) => {
+        fetchThoughts();
+      });
+  };
+
+  const onLikesIncrease = (thoughtId) => {
```

Comment:
as a reminder of coding convention again :) if the function is going to be passed as props than its a convention to start it with 'handle' instead of 'on'